kofta/src/vscode-webview/components/CreateRoomModal.tsx

```diff
@@ -11,99 +11,125 @@ import { InputField } from "./form-fields/InputField";
+	description: currentDescription,
+	isPrivate,
+	edit,
 }) => {
-  const history = useHistory();
-  return (
-    <Modal isOpen onRequestClose={onRequestClose}>
-      <Formik<{
-        name: string;
-        privacy: string;
-      }>
-        initialValues={{
-          name: "",
-          privacy: "public",
-        }}
-        validateOnChange={false}
-        validateOnBlur={false}
-        validate={({ name }) => {
-          const errors: Record<string, string> = {};
+	const history = useHistory();
+	return (
+		<Modal isOpen onRequestClose={onRequestClose}>
+			<Formik<{
+				name: string;
+				privacy: string;
+				description?: string;
+			}>
+				initialValues={{
+					name: currentName || "",
+					description: currentDescription || "",
+					privacy: isPrivate ? "private" : "public",
+				}}
+				validateOnChange={false}
+				validateOnBlur={false}
+				validate={({ name }) => {
+					const errors: Record<string, string> = {};
 
-          if (name.length < 2) {
-            return {
-              name: "min length 2",
-            };
-          }
+					if (name.length < 2) {
+						return {
+							name: "min length 2",
+						};
+					}
 
-          return errors;
-        }}
-        onSubmit={async ({ name, privacy }) => {
-          const resp = await wsFetch<any>({
-            op: "create_room",
-            d: { name, privacy },
-          });
-          if (resp.error) {
-            showErrorToast(resp.d);
-            return;
-          } else if (resp.room) {
-            const { room } = resp;
-            console.log("new room voice server id: " + room.voiceServerId);
-            useRoomChatStore.getState().clearChat();
-            wsend({ op: "get_current_room_users", d: {} });
-            history.push("/room/" + room.id);
-            useCurrentRoomStore
-              .getState()
-              .setCurrentRoom(() => roomToCurrentRoom(room));
-          }
-          onRequestClose();
-        }}
-      >
-        {({ setFieldValue, values, isSubmitting }) => (
-          <Form>
-            <InputField
-              name="name"
-              maxLength={60}
-              placeholder="room name"
-              autoFocus
-            />
-            <div className={`grid mt-8 items-start grid-cols-1`}>
-              <select
-                className={`border border-simple-gray-3c`}
-                value={values.privacy}
-                onChange={(e) => {
-                  const v = e.target.value;
-                  setFieldValue("privacy", v);
-                }}
-              >
-                <option value="public" className={`bg-simple-gray-3c`}>
-                  public
-                </option>
-                <option value="private" className={`bg-simple-gray-3c`}>
-                  private
-                </option>
-              </select>
-            </div>
+					return errors;
+				}}
+				onSubmit={async ({ name, privacy, description }) => {
+					if (edit) {
+						wsend({
```

Comment:
switch this to use `wsFetch`, see what the modal on `staging` looks like for example